### PR TITLE
[RLlib] DreamerV3: Activate simple compilation- and model architecture tests.

### PR DESCRIPTION
**Pull request by sven1977**
DreamerV3 cleanups and documentation:
* Activate simple compilation- and model architecture tests.
* Fix bugs related to cont. actions.
* Add algo to docs (including benchmark results).
* finish DreamerV3 README.md (including results)

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py [hunk 1]
```diff
@@ -53,40 +53,39 @@ def configure_optimizers_for_module(
...
             module_id=module_id,
             optimizer_name="world_model",
             optimizer=optim_world_model,
             params=params_world_model,
+            lr_or_lr_schedule=hps.world_model_lr,
         )
 
         # Actor optimizer.
-        optim_actor = tf.keras.optimizers.Adam(learning_rate=hps.actor_lr, epsilon=1e-5)
+        optim_actor = tf.keras.optimizers.Adam(epsilon=1e-5)
```
Comment (kouroshHakha) [inline rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py:68]: Why are we hardcoding the lr as opposed to reading it from config with this default value?

File: rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py [hunk 2]
```diff
@@ -53,40 +53,39 @@ def configure_optimizers_for_module(
...
             optimizer=optim_actor,
             params=params_actor,
+            lr_or_lr_schedule=hps.actor_lr,
         )
 
         # Critic optimizer.
-        optim_critic = tf.keras.optimizers.Adam(
-            learning_rate=hps.critic_lr, epsilon=1e-5
-        )
+        optim_critic = tf.keras.optimizers.Adam(epsilon=1e-5)
```
Comment (kouroshHakha) [inline rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py:80]: Same thing?

**Review by kouroshHakha — COMMENTED** [review]
@sven1977 can you breakdown your pr into three prs?

1. Only docs (no source code)
2. Dreamer only code
3. Code changes for ci/regression test

So if anything has to be reverted or modified it's isolated from the rest of them.

File: rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py (hunk 1 above)
Comment (sven1977) [reply rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py:68]: We are setting the lr anyways to the configured value (or schedule) a few lines down below. So there is no need to set it via the constructor.

File: rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py (hunk 2 above)
Comment (sven1977) [reply rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py:80]: same, see my explanation above.

File: rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py (hunk 1 above)
Comment (kouroshHakha) [reply rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py:68]: Oh sorry I thought we were setting lr to 1e-5. I miss read the code. 👍

**Pull request closed without merging**